### PR TITLE
[CARBONDATA-2133] Fixed Exception displays after performing select query on newly added Boolean Type

### PR DESCRIPTION
**Pull request by anubhav100**
**Problem** : In Restructure util and  RestructureBasedVectorResultCollector to get the default value of a measure type the case for boolean data type was missing,and in DataTypeUtil to store default value in bytes case of boolean data type was missing

**Solution**:Add the Required Cases

Be sure to do all of the following checklist to help us incorporate
your contribution quickly and easily:

- Any interfaces changed?
NA

 Any backward compatibility impacted?
NA

- Document update required?
NA

-Testing done
manual testing

Please provide details on
- Whether new unit test cases have been added or why no new tests are required?
yes

- How it is tested? Please attach test report.
mvn  -Pspark-2.1 clean install is passing

- Is it a performance related change? Please attach the performance test report.
no

- Any additional information to help reviewers in testing this change.
no
 For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.
NA

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/3533/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/2297/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/3375/

File: core/src/main/java/org/apache/carbondata/core/util/DataTypeUtil.java [hunk 1]
```diff
@@ -720,21 +721,22 @@ private static String parseStringToBigDecimal(String value, ColumnSchema columnS
     return null;
   }
 
+  public static DataTypeConverter getDataTypeConverter() {
+    if (converter == null) {
+      converter = new DataTypeConverterImpl();
+    }
+    return converter;
+  }
+
```
Comment (manishgupta88) [inline core/src/main/java/org/apache/carbondata/core/util/DataTypeUtil.java]: This change does not belong to this PR...the code is already present in the same class...kindly remove all the changes that does not belong to this PR

File: integration/spark2/src/test/scala/org/apache/spark/carbondata/restructure/AlterTableValidationTestCase.scala [hunk 2]
```diff
@@ -544,6 +546,13 @@ class AlterTableValidationTestCase extends Spark2QueryTest with BeforeAndAfterAl
     sql("drop table if exists restructure1")
     sql("drop table if exists restructure")
   }
+test("test alter command for boolean data type with correct default measure value") {
+  sql("create table testalterwithboolean(id int,name string) stored by 'carbondata' ")
+  sql("insert into testalterwithboolean values(1,'anubhav')  ")
+  sql(
+    "alter table testalterwithboolean add columns(booleanfield boolean) tblproperties('default.value.booleanfield'='true')")
+  checkAnswer(sql("select * from testalterwithboolean"),Seq(Row(1,"anubhav",true)))
+}
```
Comment (manishgupta88) [inline integration/spark2/src/test/scala/org/apache/spark/carbondata/restructure/AlterTableValidationTestCase.scala]: Please add one more test case for boolean type where the default value is not provided

**Comment by manishgupta88** [comment]
@anubhav100 ... In your PR description above no need to mention about the code details...that will be reviewed as part of your PR....kindly remove the code details from PR description and follow the PR description template

File: integration/spark2/src/test/scala/org/apache/spark/carbondata/restructure/AlterTableValidationTestCase.scala (hunk 2 above)
Comment (anubhav100) [reply integration/spark2/src/test/scala/org/apache/spark/carbondata/restructure/AlterTableValidationTestCase.scala]: done

**Comment by anubhav100** [comment]
@manishgupta88 i have done changes please check

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/3742/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/2502/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/3541/

**Comment by anubhav100** [comment]
retest sdv please

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/3543/

**Comment by anubhav100** [comment]
retest sdv please

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/3544/

**Comment by anubhav100** [comment]
retest sdv please

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/3589/

**Comment by manishgupta88** [comment]
LGTM
